src/Microsoft.OpenApi.Readers/OpenApiReaderSettings.cs

```diff
@@ -49,5 +49,9 @@ public class OpenApiReaderSettings
         /// </summary>
         public ValidationRuleSet RuleSet { get; set; } = ValidationRuleSet.GetDefaultRuleSet();
 
+        /// <summary>
+        /// URL where relative references should be resolved from if the description does not contain Server definitions
+        /// </summary>
+        public Uri BaseUrl { get; set; } = new Uri("https://example.org/");
```

Comment:
> Uri("https://example.org/"); [](start = 46, length = 29)

The spec says this: 

[https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#fixed-fields](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#fixed-fields)

 An array of Server Objects, which provide connectivity information to a target server. If the servers property is not provided, or is an empty array, the default value would be a Server Object with a url value of /.

Reply:
Actually, that might be unrelated. Let me read through more, and I'll readdress this.

---
In reply to: [193214821](https://github.com/Microsoft/OpenAPI.NET/pull/267#discussion_r193214821) [](ancestors = 193214821)

Reply:
OK. I finished looking through the whole PR. I don't think we should use an arbitrary URL here. 

I'd handle this inside the deserializer itself (see my other comment there about special cases). It seems a bit confusing to have a baseUrl in the reader/parsingContext, especially the one that's not given by the caller.

---
In reply to: [193215807](https://github.com/Microsoft/OpenAPI.NET/pull/267#discussion_r193215807) [](ancestors = 193215807,193214821)

Reply:
I can leave this null by default and fix the MakeServer stuff, but we still need a way for the user to specify the URL where the doc came from.

Reply:
Sounds good.

---
In reply to: [193230553](https://github.com/Microsoft/OpenAPI.NET/pull/267#discussion_r193230553) [](ancestors = 193230553)